integration-tests/profiler/profiler.spec.js

```diff
@@ -454,6 +454,33 @@ describe('profiler', () => {
       assert.equal(endpoints.size, 3, encoded)
     })
 
+    it('number of events is limited', async () => {
+      const threads = 1
+      const uploadPeriodSec = 10
+      const samplingPeriodMs = 1000 / 99
+      // Recreates getMaxSamples from packages/dd-trace/src/profiling/profilers/events.js
+      const maxEvents = Math.floor((threads + 2) * uploadPeriodSec * 1000 / samplingPeriodMs)
+
+      // Test will try to force maxEvents + 10 events, we'll assert that only maxEvents are captured.
```

Comment:
What do you mean by "try"? Isn't it guaranteed to be at least maxEvents + 10 due to the test events being emitted?

Reply:
That's unfortunate wording indeed. It's just my inner pessimist allowing for any test having a chance to fail.